### PR TITLE
fix(progress-indicator): add support for RTL

### LINKED ISSUE #471: RTL: Progress Indicator
Screen Shot 2019-08-08 at 11 04 55 AM

Screen Shot 2019-08-08 at 11 05 59 AM

Screen Shot 2019-08-08 at 11 06 51 AM

- Missing padding between circle icon and accompanying text
- "Presentation" version has extra space between steps for some reason (also in LTR)
- "Presentation vertical" version needs to have the secondary label left-aligned

### PR DESCRIPTION
**Pull request by enricoberti**
Closes #471

**Summary**

- Pass on the horizontal and vertical progress indicators

**Change List (commits, features, bugs, etc)**

- added specific scss + unspecific for the skeleton which is inherited directly from Carbon

**Acceptance Test (how to verify the PR)**

image

### CODE REVIEW

File: src/components/ProgressIndicator/ProgressIndicator.jsx [hunk 1]
```diff
@@ -105,10 +109,13 @@ const ProgressIndicator = ({
...
   const currentStep = matchingIndex > -1 ? matchingIndex : 0;
-
   return (
     <StyledProgressIndicator
-      className={[className, isVerticalMode ? 'bx--progress--vertical' : ''].join(' ')}
+      className={[
+        className,
+        isVerticalMode ? 'bx--progress--vertical' : '',
+        `${iotPrefix}--progress-indicator`,
+      ].join(' ')}
```
Comment (tay1orjones) [inline src/components/ProgressIndicator/ProgressIndicator.jsx]: The `bx` prefix can't be hardcoded as it's configurable. IIRC we can import it from `settings`. 

Also lets push for consistency and use `classnames` in lieu of `.join(' ')`:

```suggestion
      className={classNames({
        className,
        `${iotPrefix}--progress-indicator`
        `${prefix}--progress--vertical`: isVerticalMode,
      })}
```

File: src/components/ProgressIndicator/_progress-indicator.scss [hunk 2]
```diff
@@ -1,5 +1,36 @@
 @import '~carbon-components/scss/components/progress-indicator/progress-indicator';
+@import '../../globals/vars';
 
-.bx--skeleton .bx--progress-step {
-  flex: unset;
+.#{$iot-prefix}--progress-indicator {
+  .bx--skeleton .bx--progress-step {
```
Comment (tay1orjones) [inline src/components/ProgressIndicator/_progress-indicator.scss]: Same thing here with the hardcoded prefix:
 ```suggestion
  .#{prefix}--skeleton .#{prefix}--progress-step {
```

File: src/components/ProgressIndicator/_progress-indicator.scss [hunk 3]
```diff
@@ -1,5 +1,36 @@
...
+    flex: unset;
+  }
+}
+
+html[dir='rtl'] {
+  .#{$iot-prefix}--progress-indicator {
+    .bx--progress-step svg {
+      margin-right: 0;
+      margin-left: 0.5rem;
```
Comment (tay1orjones) [inline src/components/ProgressIndicator/_progress-indicator.scss]: Where possible, these can be modified to use the [carbon spacing scale variables](https://www.carbondesignsystem.com/guidelines/spacing#spacing-scale)

```suggestion
      margin-left: $spacing-03;
```

**Comment by enricoberti** [comment]
@tay1orjones updated this too! thanks!

**Pull request merged**